modules/swagger-core/src/main/java/io/swagger/v3/core/util/ModelDeserializer.java

```diff
@@ -77,6 +78,10 @@ public Schema deserialize(JsonParser jp, DeserializationContext ctxt)
                 }
             } else if (node.get("$ref") != null) {
                 schema = new Schema().$ref(node.get("$ref").asText());
+                Map<String,Object> extensions = getExtensions(node);
+                if(extensions != null && extensions.size() > 0) {
+                  schema.setExtensions(extensions);
+                }
```

Comment:
deserialize extensions when used with $ref in schemas.
We dont use this functionality in the platform, but its good to have the extensions feature supported in both ser/deser

Reply:
Implementing for just one usecase (ser), seemed incomplete